### ISSUE TITLE
Slack notification delay after new issue arrived to sentry

### ISSUE THREAD

**Issue by maylow** [getsentry/sentry#70285]
### Self-Hosted Version

24.4.1

### CPU Architecture

x86_64

### Docker Version

20.10.24

### Docker Compose Version

2.23.3

### Steps to Reproduce

1. Create alert to send slack notification when new issue is created.
2. Send issue via sentry-cli.
3. Sentry gets an issue but not mark it as new issue for 5 minutes.

![SCR-20240430-jift](https://github.com/getsentry/self-hosted/assets/11584131/242a438b-a985-4d70-8a8a-cde6e8f996f4)

4. After 5 minutes, it gets mark new issue and send issue to slack.

<img width="692" alt="SCR-20240430-jisg" src="https://github.com/getsentry/self-hosted/assets/11584131/86be8ad6-b673-49d7-ade1-175d6edf6509">

### Expected Result

1. Create alert to send slack notification when new issue is created.
2. Send issue via sentry-cli.
3. Sentry gets an issue, mark it as new issue and send it to slack.

### Actual Result

It is regression of the issue: https://github.com/getsentry/self-hosted/issues/2034
The issue showed up after upgrade from 24.3.0 -> 24.4.0 with continue to 24.4.1

I also find the same reason where kafka has LAG on post-process events:
```
GROUP                  TOPIC           PARTITION  CURRENT-OFFSET  LOG-END-OFFSET  LAG             CONSUMER-ID                                  HOST            CLIENT-ID
post-process-forwarder generic-events  0          478291          478291          0               rdkafka-b5bbee3d-650c-465f-aeb5-b1b2b78eb9a9 /172.18.0.43    rdkafka
post-process-forwarder events          0          9288607         9288608         1               rdkafka-a7f67d85-627b-4fae-ba46-c6e03a7bb3d4 /172.18.0.51    rdkafka
post-process-forwarder transactions    0          186570290       186570290       0               rdkafka-482098cd-d1ab-46fd-8ee6-b9ae3dcf3aea /172.18.0.42    rdkafka
```

**Comment by maylow** [comment]
do we have any update?

**Comment by scttcper** [comment]
What does the alert rule being triggered look like? What do the alert settings look like for that project? 
![image](https://github.com/getsentry/sentry/assets/1400464/627ce1ba-068d-4c0b-a5b6-b9389e46088b)

The default delay is 5 minutes.

**Comment by maylow** [comment]
Hello @scttcper our rule look like
![image](https://github.com/getsentry/sentry/assets/11584131/f7dc8918-3da6-471b-8e70-e52895adfa09)

**Comment by scttcper** [comment]
what does the issue alert look like?

**Comment by maylow** [comment]
You can see that new issue has right time and showed up ok but it is ongoing.
![SCR-20240605-hkgw](https://github.com/getsentry/sentry/assets/11584131/56598b3b-4226-4a7e-9777-116be5bdca8a)

It is marked as NEW after 5 minutes.

![SCR-20240605-hmbd](https://github.com/getsentry/sentry/assets/11584131/c21c4b2f-fdd0-496e-85a7-0052de7a77dc)

**Comment by leeandher** [comment]
I have not been able to reproduce this issue, are you able to share the alert itself (e.g. the when/if conditions and actions like sending to a slack channel)?

**Comment by hubertdeng123** [comment]
@leeandher I think the main thing here is that when issues come in for the first time, they're marked as `Ongoing` for some reason, which is only changed to `New` when new events are ingested by kafka. For self-hosted specifically, it's a larger issue since events are ingested at a far lower rate than in SaaS.

**Comment by maylow** [comment]
Thats true. When the issue come in then it is marked as `Ongoing`. `Ongoing` status will be changed once another issue come in.

I send issue via `sentry-cli` to show you what is going on:
![first](https://github.com/getsentry/sentry/assets/11584131/51558914-36e9-4d3f-8a32-988ddd6f4980)
- Issue come in and it is marked as `Ongoing`.

I send new issue:
![second](https://github.com/getsentry/sentry/assets/11584131/a0f854a1-7b3d-4ecd-97b5-fdc9c03ca3bd)
- Both are marked as `Ongoing`.

After a few seconds:
![third](https://github.com/getsentry/sentry/assets/11584131/e694f5f6-d0f4-44c4-a108-ec57fa872b7f)
- Oldest issue change status from `Ongoing` to `New` -> sentry will send issue to slack.

Issue that come in to sentry should be immediately marked as `New` which will trigger alert and send message to slack.

**Comment by leeandher** [comment]
Thanks for the additional context! I'll be taking a look at this again tomorrow morning to reproduce it and look into why they're being incorrectly categorized.

**Comment by leeandher** [comment]
Alright I can confirm [where the issue is coming from](https://github.com/getsentry/sentry/blob/6775cceb9c6ef43b8aff93107137502da512792d/src/sentry/models/group.py#L986) (thanks @snigdhas) but we'll have to be a little precise with how we fix it as there are a few downstream affects that will cause issues if we simply update this line. 

We're going to factor this issue into a larger scope fix to address some of these status inconsistencies, but I'm not able to provide a concrete timeline on when to expect a fix. Thanks again for all the context and for reporting this!

**Comment by simPod** [comment]
Hello, do we have an update? Thanks

**Comment by snigdhas** [comment]
Hi @simPod, thanks for checking in. We're actively working on the fix for this as part of #75454. The fix should be rolled out by the end of the month.

**Comment by cusher** [comment]
Hi @snigdhas, could you confirm whether this is supposed to be fixed in 24.10.0? We are still hitting this issue after updating, and just want to verify that the expected fix has actually been rolled out already before looking further into what's going wrong. Thank you!
